solr/core/src/java/org/apache/solr/cloud/api/collections/DeleteCollectionCmd.java

```diff
@@ -168,7 +168,7 @@ public void call(ClusterState state, ZkNodeProps message, NamedList<Object> resu
       }
 
       // delete related config set iff: it is auto generated AND not related to any other collection
-      String configSetName = coll.getConfigName();
+      String configSetName = coll == null ? null : coll.getConfigName();
```

Comment:
IntelliJ helpfully pointed out an issue nearby this PR with a trivial fix.  Basically, it might not have been possible to delete a non-existent collection but now it might be.  (couldn't find a test for this)